schema_based_on_diagram.sql

```diff
@@ -0,0 +1,61 @@
+-- create database clinic
+CREATE DATABASE clinic;
+
+-- create patients table
+CREATE TABLE patients (
+    id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    name VARCHAR(100) NOT NULL,
+    date_of_birth DATE NOT NULL
+);
+
+-- create treatments table
+ CREATE TABLE treatments (
+     id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+     type VARCHAR(100) NOT NULL,
+     name VARCHAR(120) NOT NULL
+);
+
+-- create medical_histories table
+CREATE TABLE medical_histories(
+     id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+     admitted_at timestamp,
+     patient_id INT REFERENCES patients (id),
+     status VARCHAR(100) NOT NULL
+);
+
+
+-- create invoices table
+CREATE TABLE invoices (
+    id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    total_amount DECIMAL NOT NULL,
+    generated_at TIMESTAMP,
+    payed_at TIMESTAMP,
+    medical_history_id int REFERENCES medical_histories(id)
+);
+
+
+-- create invocie_items table
+CREATE TABLE invocie_items (
+    id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    unique_price DECIMAL NOT NULL,
```

Comment:
- [ ] From the uml diagram show below in the image, your table is expected to have a `unit_price` not `unique_price` as you have used in the highlighted line, please kindly fix this :+1: :sparkles: 

![un](https://github.com/MozamelJawad/clinic_db/assets/89055502/f477307b-cbda-451e-9051-35bb2907d209)